### PR TITLE
feat: Add group_by_state support to expenses stats API

### PR DESCRIPTION
**Pull request by devadathanmb**
### Description
Please add PR description here, add screenshots if needed

image

## Clickup
https://app.clickup.com/1864988/v/l/li/901604687952

### CODE REVIEW

File: src/components/schemas/expense.yaml [hunk 1]
```diff
@@ -1678,29 +1678,81 @@ expense_delete_in:
 
 
 expenses_stats_in:
+  type: object
+  required:
+    - data
   properties:
     query_params:
-      nullable: False
+      nullable: false
```
Comment (KirtiGautam) [inline src/components/schemas/expense.yaml:1686]: Is this case insensitive? If not, then was it wrong before? Can you please check this?

File: src/components/schemas/expense.yaml [hunk 2]
```diff
@@ -1678,29 +1678,81 @@ expense_delete_in:
...
-      description: |
-        This field represents the total number of expenses that match the criteria specified in the query_params within the body of the request.
-      example: 1     
-    total_amount:
-      allOf:
-        - $ref: './fields.yaml#/amount'
-      description: |
-        This field indicates the sum of amount of expenses that match the criteria specified in the query_params within the body of the request.
-      example: 100
+  type: array
```
Comment (KirtiGautam) [inline src/components/schemas/expense.yaml:1693]: Shouldn't this be `OneOf`? Case1: not group by and Case2: group by?

File: src/components/schemas/expense.yaml (hunk 1 above)
Comment (devadathanmb) [reply src/components/schemas/expense.yaml]: [Swagger docs](https://swagger.io/docs/specification/v3_0/data-models/data-types/#null) use lowercase all over the place. Both seem to work though.

File: src/components/schemas/expense.yaml (hunk 2 above)
Comment (devadathanmb) [reply src/components/schemas/expense.yaml]: Updated

**Review by KirtiGautam — APPROVED** [review]
LGTM 🍏

**Pull request merged**